modules/checklist/module.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Elementor\Modules\Checklist;
+
+use Elementor\Core\Base\Module as BaseModule;
+use Elementor\Core\Experiments\Manager;
+use Elementor\Core\Isolation\Wordpress_Adapter;
+use Elementor\Plugin;
+use Elementor\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Module extends BaseModule {
+	const EXPERIMENT_ID = 'launchpad-checklist';
+	const DB_OPTION_KEY = 'e-checklist';
+
+	private $user_progress = null;
+
+	private Steps_Manager $steps_manager;
+
+	private Wordpress_Adapter $wordpress_adapter;
+
+	/**
+	 * @param ?Wordpress_Adapter $wordpress_adapter
+	 *
+	 * @return void
+	 */
+	public function __construct( $wordpress_adapter = null ) {
+		$this->wordpress_adapter = $wordpress_adapter ?? new Wordpress_Adapter();
+		parent::__construct();
+
+		$this->register_experiment();
+
+		if ( ! $this->is_experiment_active() ) {
+			return;
+		}
+
+		$this->setup();
+	}
+
+	/**
+	 * Get the module name.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return 'e-checklist';
+	}
+
+	/**
+	 * Checks if the experiment is active
+	 *
+	 * @return bool
+	 */
+	public function is_experiment_active(): bool {
+		return Plugin::$instance->experiments->is_feature_active( self::EXPERIMENT_ID );
+	}
+
+	/**
+	 * Gets user's progress from db
+	 *
+	 * @return array {
+	 *      @type bool $is_hidden
+	 *      @type int $last_opened_timestamp
+	 *      @type array $steps {
+	 *          @type string $step_id => {
+	 *              @type bool $is_marked_done
+	 *              @type bool $is_completed
+	 *          }
+	 *      }
+	 *  }
+	 */
+	public function get_user_progress_from_db() {
+		return json_decode( get_option( self::DB_OPTION_KEY ), true );
+	}
+
+	/**
+	 * Using the step's ID, get the progress of the step should it exist
+	 *
+	 * @param $step_id
+	 *
+	 * @return null|array {
+	 *      @type bool $is_marked_done
+	 *      @type bool $is_completed
+	 *  }
+	 */
+	public function get_step_progress( $step_id ) {
+		foreach ( $this->user_progress['steps'] as $id => $step ) {
```

Comment:
can we just check for `$this->user_progress['steps'][$step_id]` ?

Reply:
Yes you're right